### ISSUE TITLE
curl gets 500 regardless of grpc error code used in HttpsError

### ISSUE THREAD

**Issue by ekalosak** [firebase/firebase-functions-python#129]
I'm raising `https_fn.HttpsError` with various `code=https_fn.FunctionsErrorCode.XYZ` from within an `@https_fn.on_request()`-decorated function to test error paths. But if I throw e.g. ostensibly a 400 (`FunctionsErrorCode.NOT_FOUND`) the caller gets a 500. It doesn't matter what type of error I throw. Is this intentional for `on_request`? I'd love to have more granular HTTP errors.

Anyway, thanks very much for a lovely Python3 SDK for FB Functions - it's a super convenient service.

**Comment by ekalosak** [comment]
For reference, I'm expecting the behavior described in [this doc](https://firebase.google.com/docs/functions/callable?gen=2nd#handle_errors). Maybe I misread it... I'll give it another close reading but I figured it's good to get this reported.

**Comment by ekalosak** [comment]
Yeah interestingly enough `@https_fn.on_call` handles the error codes as I'd expect. It's odd to see the ∆ between `on_call` and `on_request` error handling.

**Comment by exaby73** [comment]
Hello @ekalosak. Could you provide a short code sample that I may use to reproduce this issue? Also, do include the contents of your `requirements.txt` and python version as well.
